### PR TITLE
Feature/issue42 current api calling functions

### PR DESCRIPTION
**Pull request by moewiz**
TODO
---

- [x] Call GET method to retrieve current position data.
- [x] Call POST method to update the current position.
- [x] Create store, reducer, saga & actions for handling this API call.

### CODE REVIEW

File: src/layouts/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -45,20 +47,21 @@ class Sidebar extends React.PureComponent<SidebarProps & RouteComponentProps> {
 
   public showTable = (tagName: string, tabName: string) => {
     const { history } = this.props;
-    const clientId = '123456';
+    const clientId = '5';
```
Comment (dinhnguyentrunghoa) [inline src/layouts/Sidebar/Sidebar.tsx]: Should have a TODO comment to note that this id will get from another incompleted API

File: src/enums/client.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export interface Position {
```
Comment (dinhnguyentrunghoa) [inline src/enums/client.ts]: Check again the structure. It makes confuse when import an interface in enum file

File: src/reducers/client/clientTypes.ts [hunk 3]
```diff
@@ -56,7 +57,7 @@ export const getDefaultTagList = () => {
 export const defaultClientState: ClientState = {
   clients: [
     {
-      clientId: '123456',
+      clientId: 5,
```
Comment (dinhnguyentrunghoa) [inline src/reducers/client/clientTypes.ts]: Get the client from redux.

File: src/layouts/Sidebar/Sidebar.tsx (hunk 1 above)
Comment (dinhnguyentrunghoa) [inline src/layouts/Sidebar/Sidebar.tsx]: Get the client from redux.

File: src/layouts/Sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -45,20 +47,21 @@ class Sidebar extends React.PureComponent<SidebarProps & RouteComponentProps> {
...
 
     history.push(`/client/${clientId}/${tagName}/${tabName}`);
   }
 
   public selectClient = () => {
     const { history } = this.props;
-    const clientId = '123456';
+    const clientId = '5';
```
Comment (dinhnguyentrunghoa) [inline src/layouts/Sidebar/Sidebar.tsx]: Get the client from redux.

File: src/sagas/client/clientSaga.ts [hunk 5]
```diff
@@ -1,20 +1,53 @@
...
+          yield put({
+            type: ClientActionTypes.FETCH_DATA_ENTRY_SUCCESS,
+            payload: {
+              clientId,
+              tagName,
+              tabName,
+              dataEntry: response.data.data,
+            },
+          });
+        }
```
Comment (dinhnguyentrunghoa) [inline src/sagas/client/clientSaga.ts]: What happened if the Axios call is failed? We should throw an error here

Comment (moewiz) [reply src/sagas/client/clientSaga.ts]: @dinhnguyentrunghoa If the called API has failed, Axios will execute the rejected case and the `catch` block will handle this situation.

File: src/reducers/client/clientTypes.ts (hunk 3 above)
Comment (moewiz) [reply src/reducers/client/clientTypes.ts]: For this moment, we set the default value for the list of clients.

**Pull request merged**